### ISSUE TITLE
Different API base support

### ISSUE THREAD

**Issue by Lachine1** [ThioJoe/Full-Stack-AI-Meme-Generator#12]

**Comment by Lachine1** [comment]
for openai btw

**Comment by ThioJoe** [comment]
I don't undestand

**Comment by Lachine1** [comment]
> I don't undestand

openai.api_base = "https://example.api.com/"

**Comment by ThioJoe** [comment]
I see, I'll look into adding that

**Comment by NavaShield** [comment]
agreed, there are openai alternatives which use the same url structure as openai so there could be a commented out setting in the settings.ini file to enable a custom base url.
```ini
# Allows use of custom base urls.
# BaseURL = https://api.openai.com/v1
```
or something like that

**Comment by ThioJoe** [comment]
Just so you know though, the script uses the OpenAI python package, not URL endpoints directly. So might not be possible in the current state.

**Comment by Lachine1** [comment]
you should use dall-e for image gen too

**Comment by NavaShield** [comment]
> Just so you know though, the script uses the OpenAI python package, not URL endpoints directly. So might not be possible in the current state if the goal is to use a different service altogether.

```py
import openai

openai.api_key = 'catto_key_************************'
openai.api_base = 'https://api.cattto.repl.co/v1'
```

(code stolen from a chatgpt alternative but yeah)

**Comment by arcaela** [comment]
If/when custom base URL support lands, Arko Studio (https://arko.arcaelas.com) could be an option for users who want to generate memes without paying for OpenAI API credits. It provides free API keys (no credit card) and is compatible with the OpenAI Python SDK — just set base_url. It also supports image generation through the same API.
